src/Facility.Definition/ServiceInfo.cs

```diff
@@ -155,9 +165,85 @@ ServiceMemberInfo DoExcludeTag(ServiceMemberInfo member)
+					var patternParameter = validateAttribute.TryGetParameter("pattern");
+
+					if (lengthParameter == null && patternParameter == null)
+						field.AddValidationError(ServiceDefinitionUtility.CreateMissingAttributeParameterError(validateAttribute, "length' or 'pattern"));
+					if (lengthParameter != null && !s_rangeRegex.IsMatch(lengthParameter.Value))
+						field.AddValidationError(ServiceDefinitionUtility.CreateInvalidAttributeValueError(validateAttribute.Name, lengthParameter));
+					if (patternParameter != null && !s_patternRegex.IsMatch(patternParameter.Value))
+						field.AddValidationError(ServiceDefinitionUtility.CreateInvalidAttributeValueError(validateAttribute.Name, patternParameter));
+
+					break;
+				}
+
+				case ServiceTypeKind.Double:
+				case ServiceTypeKind.Int32:
+				case ServiceTypeKind.Int64:
+				case ServiceTypeKind.Decimal:
+				{
+					foreach (var validateAttributeParameter in validateAttribute.Parameters.Where(x => x.Name != "value"))
+						field.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(validateAttribute.Name, validateAttributeParameter));
+
+					if (validateAttribute.Parameters.All(x => x.Name != "value"))
+						field.AddValidationError(ServiceDefinitionUtility.CreateMissingAttributeParameterError(validateAttribute, "value"));
+
+					foreach (var parameter in validateAttribute.Parameters.Where(x => !s_rangeRegex.IsMatch(x.Value)))
+						field.AddValidationError(ServiceDefinitionUtility.CreateInvalidAttributeValueError(validateAttribute.Name, parameter));
+					break;
+				}
+
+				case ServiceTypeKind.Bytes:
+				case ServiceTypeKind.Array:
+				case ServiceTypeKind.Map:
+				{
+					foreach (var validateAttributeParameter in validateAttribute.Parameters.Where(x => x.Name != "count"))
+						field.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeParameterError(validateAttribute.Name, validateAttributeParameter));
+
+					if (validateAttribute.Parameters.All(x => x.Name != "count"))
+						field.AddValidationError(ServiceDefinitionUtility.CreateMissingAttributeParameterError(validateAttribute, "count"));
+
+					foreach (var parameter in validateAttribute.Parameters.Where(x => !s_rangeRegex.IsMatch(x.Value)))
+						field.AddValidationError(ServiceDefinitionUtility.CreateInvalidAttributeValueError(validateAttribute.Name, parameter));
+
+					break;
+				}
+
+				default:
+					field.AddValidationError(ServiceDefinitionUtility.CreateUnexpectedAttributeError(validateAttribute));
+					break;
+			}
+		}
 		private protected override IEnumerable<ServiceElementInfo> GetExtraChildrenCore() => Members;
 
 		private readonly Dictionary<string, ServiceMemberInfo> m_membersByName;
 		private readonly Dictionary<string, ServiceTypeInfo> m_typesByName;
+
+		private static readonly Regex s_patternRegex = new(@"^"".+""$");
+		private static readonly Regex s_rangeRegex = new(@"(?:\d+\.\d+)?\.\.(?:\d+\.\d+)?");
```

Comment:
These are rough approximations of what would eventually be required.
The pattern regex is probably wrong as-is. It shouldn't need to match double quotes.
The range regex is definitely wrong.

Reply:
The `..` should not be required, e.g. `count: 2` would require exactly two items.